solr/core/src/java/org/apache/solr/api/AnnotatedApi.java

```diff
@@ -225,7 +228,11 @@ public void call(SolrQueryRequest req, SolrQueryResponse rsp) {
       if (Modifier.isPublic(method.getModifiers())) {
         this.command = command;
         this.obj = obj;
-        this.method = method;
+        try {
+          this.method = MethodHandles.publicLookup().unreflect(method);
+        } catch (IllegalAccessException e) {
+          throw new RuntimeException("Unable to unlookup method");
```

Comment:
Lets throw SolrException here.

Reply:
Unrelated to to this PR (i.e. comment holds for previous version of the code), maybe `InterruptedException` should be caught separately and `Thread.currentThread().interrupt()` be called.

Reply:
unlookup? (did you mean unreflect?)

Reply:
The exception text should not mention internal details. It should just say that the method cannot be called, as it's not accessible (non public, class not visible, different classloader,...)